src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/LdapRegistrationDAO.scala

```diff
@@ -156,4 +157,17 @@ class LdapRegistrationDAO(
 
   override def setGoogleSubjectId(userId: WorkbenchUserId, googleSubjectId: GoogleSubjectId, samRequestContext: SamRequestContext): IO[Unit] =
     executeLdap(IO(ldapConnectionPool.modify(userDn(userId), new Modification(ModificationType.ADD, Attr.googleSubjectId, googleSubjectId.value))), "setGoogleSubjectId", samRequestContext)
+
+  override def checkStatus(samRequestContext: SamRequestContext): Boolean = {
+    val ldapIsHealthy = Try {
+      ldapConnectionPool.getHealthCheck
+      val connection = ldapConnectionPool.getConnection
+      ldapConnectionPool.getHealthCheck.ensureNewConnectionValid(connection)
```

Comment:
I don't _think_ that `ensureConnectionValidAfterAuthentication` is the right method. It seems to use a BindResult https://docs.ldap.com/ldap-sdk/docs/javadoc/com/unboundid/ldap/sdk/BindResult.html which doesn't actually look like its about auth? I am not 100% certain on that, though. I couldn't find any usages of BindResult anywhere in the code base, it looks like we just pass in a user and a password when setting up the pool, the once.

I would think that the thing to test that might be an integration test, not as part of our status check, because those values shouldn't be set anywhere, just once when the service stands up.

Reply:
Seconding Sky, I'm not sure how we'd go about getting the `BindResult`. It'd be great to have our status check make sure we can auth with LDAP, but `ensureConnectionValidAfterAuthentication` may not be the answer